### DISCUSSION TITLE
make v doc more useful

### DISCUSSION THREAD

**Discussion by bakul** [vlang/v#14617] [Ideas]
Allow `v doc <module>.<component>` to display the component declaration / definition and any comment above it. Compare with, for example `go doc os.ReadDir`. The <module> can a dot separated list. Currently `v doc os File` for examples FIle struct type and its methods. This can continue work but `v doc os.File` should also work.

**Comment by splace** [comment]
+1

i have scripts to fire doc from any highlighted code, not having "." separator, like OP wants, breaks this. (developed for Go)

also, making a lot of use of doc on Go, it would also be great to have ".method" and -src (show source) together this turns the whole library into context sensitive examples.
